### PR TITLE
Fix case of Python.

### PR DESCRIPTION
**Pull request by jbampton**
Changed `python` to `Python`.

### CODE REVIEW

File: CHANGELOG.txt [hunk 1]
```diff
@@ -398,7 +398,7 @@ Misc/Internal
 - Move Dockerfile to Dockerfile.ci (#7829)
 - Generate requirements are now sorted (#8040)
 - Change name of the common environment initialization function (#7805)
-- Requirements now depend on python version (#7841)
+- Requirements now depend on Python version (#7841)
```
Comment (potiuk) [inline CHANGELOG.txt]: Those refer to commit messages that cannot be changed so we better do not change them.

File: airflow/providers/databricks/PROVIDERS_CHANGES_2020.06.24.md [hunk 2]
```diff
@@ -9,7 +9,7 @@
 | [f6bd817a3](https://github.com/apache/airflow/commit/f6bd817a3aac0a16430fc2e3d59c1f17a69a15ac) | 2020-06-16  | Introduce &#39;transfers&#39; packages (#9320)                                                                                                                             |
```
Comment (potiuk) [inline airflow/providers/databricks/PROVIDERS_CHANGES_2020.06.24.md]: Also all those PROVIDERS_CHANGES are referring to the commits so I would not change those. There are many provider's changes files below - all of them should be exempted from that change.

**Review by potiuk — CHANGES_REQUESTED** [review]
I love the change @jbampton , but I would love two things to change:

* let's not change python -> Python in the release notes where we are referring to commits that are in the history (otherwise people and tools might have difficulty in finding them)

* the general rule we have in case of such changes that we do not only fix them, but also prevent them from happening in the future. In this case it should be rather easy - just adding another check in the `.pre-commit-config.yml` - using pygrep plugin (you will find some examples in there). It should be rather easy to add a case-sensitive regular expression that finds `python` as a word excluding those CHANGES and UPDATING files.

File: airflow/utils/cli_action_loggers.py [hunk 3]
```diff
@@ -18,7 +18,7 @@
 #
 """
 An Action Logger module. Singleton pattern has been applied into this module
-so that registered callbacks can be used all through the same python process.
+so that registered callbacks can be used all through the same Python process.
```
Comment (ashb) [inline airflow/utils/cli_action_loggers.py:21]: (Strictly speaking this is correct lowercase -- it's talking about the executable)

File: airflow/utils/timezone.py [hunk 4]
```diff
@@ -112,7 +112,7 @@ def make_aware(value, timezone=None):
         raise ValueError(
             "make_aware expects a naive datetime, got %s" % value)
     if hasattr(value, 'fold'):
-        # In case of python 3.6 we want to do the same that pendulum does for python3.5
+        # In case of Python 3.6 we want to do the same that pendulum does for python3.5
```
Comment (ashb) [inline airflow/utils/timezone.py:115]: Missed `python3.5`.

File: scripts/ci/images/ci_prepare_ci_image_on_ci.sh [hunk 5]
```diff
@@ -37,7 +37,7 @@ function build_ci_image_on_ci() {
         if [[ ${GITHUB_REGISTRY_PUSH_IMAGE_TAG} != "latest" ]]; then
             python_tag_suffix="-${GITHUB_REGISTRY_PULL_IMAGE_TAG}"
         fi
-        # first we pull base python image. We will need it to re-push it after master build
+        # first we pull base Python image. We will need it to re-push it after master build
```
Comment (ashb) [inline scripts/ci/images/ci_prepare_ci_image_on_ci.sh:40]: Ones like this are referring to `docker pull python` - so not sure it makes sense here.

File: scripts/ci/libraries/_build_images.sh [hunk 6]
```diff
@@ -320,7 +320,7 @@ function build_images::print_build_info() {
 }
 
 function build_images::get_docker_image_names() {
-    # python image version to use
+    # Python image version to use
```
Comment (ashb) [inline scripts/ci/libraries/_build_images.sh:323]: Ditto -- is talking about `python:3.7` etc.

**Review by ashb — CHANGES_REQUESTED** [review]
You've missed at least one case, and some of them are referring to things where it should be kept lower case.

**Pull request closed without merging**